pkg/registries/registries.go

```diff
@@ -87,7 +77,7 @@ func (sets *mirrorSets) mergedMirrors(source string) ([]string, error) {
 		}
 		sourceInGraph := false
 		for _, m := range mirrors {
-			if m == source {
+			if m == source && len(mirrors) > 1 {
```

Comment:
This adds an ~unnecessary `source`->`source` loop, and that affects the ordering.

Consider `disjointSets` [`(source)`, `(a, source, b)`]. The correct ordering is `(a, source, b)`, but with an added `source`->`source` loop, `topoGraph.Sorted` will correctly start with `a`, but then `noIncoming` will be empty and the “break the loop” code will choose to start with `b` (because "b"<"source"), with the final ordering `(a, b, source)`.

It seems to me that, instead, `topoGraph.getNode` should be used in a new public `topoGraph.AddNode`.

Try adding
```go
{Source: "registry-f.com", Mirrors: []apicfgv1.ImageMirror{"registry-f.com"}},
{Source: "registry-f.com", Mirrors: []apicfgv1.ImageMirror{"mirror-a.com", "registry-f.com", "mirror-b.com"}},
```
to the test case.